### PR TITLE
feat(wren-ai-service): refactored API Endpoints into Independent Routers 

### PR DESCRIPTION
**Pull request by Spirizeon**
This PR fixes the issue of #735 (Work In Progress)

### CODE REVIEW

**Comment by Spirizeon** [comment]
@paopa thanks for letting me work on the issue, please review and lemme know any if any changes are needed.

**Comment by paopa** [comment]
> @paopa thanks for letting me work on the issue, please review and lemme know if any changes are needed.

Thanks for the hard work again! I will review it in these day!

File: wren-ai-service/src/web/v1/routers/asking.py
Comment (paopa) [inline wren-ai-service/src/web/v1/routers/asking.py:1]: What do you think renaming the router from asking to ask?

File: wren-ai-service/src/web/v1/routers/asking.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+- GET /asks/{query_id}/result: Retrieves the result of an ask request
+  Input: query_id in path
+  Output: AskResultResponse
+    {
+        "status": "string",
+        "result": {
+            // Result details (not specified in the provided code)
+        }
+    }
```
Comment (paopa) [inline wren-ai-service/src/web/v1/routers/asking.py:53]: gently mention as the above comment. thanks for the work!

File: wren-ai-service/src/web/v1/routers/semantics_preparations.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import uuid
+from dataclasses import asdict
+# from typing import Literal, Optional  # Unnecessary import
+
+from fastapi import APIRouter, BackgroundTasks, Depends
+# from pydantic import BaseModel  # Unnecessary import
```
Comment (paopa) [inline wren-ai-service/src/web/v1/routers/semantics_preparations.py:6]: a gently mention, the same as above comment

File: wren-ai-service/src/web/v1/routers/semantics_preparations.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+"""
+Router for handling semantics preparations requests and retrieving results.
+
+This router provides endpoints for initiating a semantics preparation operation
+and retrieving its results. It uses background tasks to process the
+semantics preparation requests asynchronously.
+
+Endpoints:
+- POST /semantics-preparations: Initiate a semantics preparation operation
+- GET /semantics-preparations/{mdl_hash}/status: Retrieve the status of a semantics preparation operation
```
Comment (paopa) [inline wren-ai-service/src/web/v1/routers/semantics_preparations.py:33]: a gently mention, the same as above comment

File: wren-ai-service/src/web/v1/routers/sql_answers.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+"""
+Router for handling SQL answer requests and retrieving results.
+
+This router provides endpoints for initiating an SQL answer operation
+and retrieving its results. It uses background tasks to process the
+SQL answer requests asynchronously.
+
+Endpoints:
+- POST /sql-answers: Initiate an SQL answer operation
+- GET /sql-answers/{query_id}/result: Retrieve the result of an SQL answer operation
```
Comment (paopa) [inline wren-ai-service/src/web/v1/routers/sql_answers.py:31]: a gently mention, the same as above comment

File: wren-ai-service/src/web/v1/routers/sql_expansions.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+"""
+Router for handling SQL expansion requests and retrieving results.
+
+This router provides endpoints for initiating an SQL expansion operation
+and retrieving its results. It uses background tasks to process the
+SQL expansion requests asynchronously.
+
+Endpoints:
+- POST /sql-expansions: Initiate an SQL expansion operation
+- GET /sql-expansions/{query_id}/result: Retrieve the result of an SQL expansion operation
```
Comment (paopa) [inline wren-ai-service/src/web/v1/routers/sql_expansions.py:32]: a gently mention, the same as above comment

File: wren-ai-service/src/web/v1/routers/sql_explanations.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+"""
+Router for handling SQL explanation requests and retrieving results.
+
+This router provides endpoints for initiating an SQL explanation operation
+and retrieving its results. It uses background tasks to process the
+SQL explanation requests asynchronously.
+
+Endpoints:
+- POST /sql-explanations: Initiate an SQL explanation operation
+- GET /sql-explanations/{query_id}/result: Retrieve the result of an SQL explanation operation
```
Comment (paopa) [inline wren-ai-service/src/web/v1/routers/sql_explanations.py:30]: a gently mention, the same as above comment

File: wren-ai-service/src/web/v1/routers/sql_regenerations.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+"""
+Router for handling SQL regeneration requests and retrieving results.
+
+This router provides endpoints for initiating an SQL regeneration operation
+and retrieving its results. It uses background tasks to process the
+SQL regeneration requests asynchronously.
+
+Endpoints:
+- POST /sql-regenerations: Initiate an SQL regeneration operation
+- GET /sql-regenerations/{query_id}/result: Retrieve the result of an SQL regeneration operation
```
Comment (paopa) [inline wren-ai-service/src/web/v1/routers/sql_regenerations.py:29]: a gently mention, the same as above comment

**Review by paopa — COMMENTED** [review]
Thanks for the work first!. overall it is good to me. I think having a few comments need to be done, then we can merge this PR. the router document you can refer to https://github.com/Canner/WrenAI/blob/cf314160f060e204a6ef229caea11ac245d76470/wren-ai-service/src/web/v1/routers/semantics_description.py#L24-L70

File: wren-ai-service/src/web/v1/routers/asking.py [hunk 8]
```diff
@@ -0,0 +1,136 @@
+import uuid
+from dataclasses import asdict
+# from typing import Literal, Optional  # Unnecessary import
+
+from fastapi import APIRouter, BackgroundTasks, Depends
+# from pydantic import BaseModel  # Unnecessary import
```
[the remark this answers is not in the document]
Comment (Spirizeon) [reply wren-ai-service/src/web/v1/routers/asking.py]: Sure.

File: wren-ai-service/src/web/v1/routers/__init__.py [hunk 9]
```diff
@@ -9,305 +9,23 @@
...
-    SQLExplanationResultRequest,
-    SQLExplanationResultResponse,
-)
-from src.web.v1.services.sql_regeneration import (
-    SQLRegenerationRequest,
-    SQLRegenerationResponse,
-    SQLRegenerationResultRequest,
-    SQLRegenerationResultResponse,
-)
+from src.web.v1.routers import relationship_recommendation, semantics_description, asking
```
[the remark this answers is not in the document]
Comment (Spirizeon) [reply wren-ai-service/src/web/v1/routers/__init__.py]: Fixed, thanks!

File: wren-ai-service/src/web/v1/routers/asking.py
Comment (Spirizeon) [reply wren-ai-service/src/web/v1/routers/asking.py:1]: Good indeed, I think renaming to `asks` would sound better.

File: wren-ai-service/src/web/v1/routers/ask_details.py [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+"""
+Router for handling SQL query details requests and retrieving results.
+
+This router provides endpoints for initiating an ask-details operation
+and retrieving its results. It uses background tasks to process the
+ask-details requests asynchronously.
+
+Endpoints:
+- POST /ask-details: Initiate an ask-details operation
+- GET /ask-details/{query_id}/result: Retrieve the result of an ask-details operation
```
[the remark this answers is not in the document]
Comment (Spirizeon) [reply wren-ai-service/src/web/v1/routers/ask_details.py:30]: Sure, I'll make changes accordingly

**Comment by Spirizeon** [comment]
> Thanks for the work first!. overall it is good to me. I think having a few comments need to be done, then we can merge this PR. the router document you can refer to
> 
> https://github.com/Canner/WrenAI/blob/cf314160f060e204a6ef229caea11ac245d76470/wren-ai-service/src/web/v1/routers/semantics_description.py#L24-L70

Thank you for the suggestions! If you can kindly review the changes now.

**Comment by paopa** [comment]
hey @Spirizeon, Thanks for the contribution again! You can DM @chilijung on Discord with https://github.com/Canner/WrenAI/issues/735 to get the swag form link.

**Pull request merged**